cpp/arcticdb/version/local_versioned_engine.hpp

```diff
@@ -273,18 +274,18 @@ class LocalVersionedEngine : public VersionedEngine {
         const WriteOptions& write_options,
         bool validate_index);
 
-    std::vector<std::pair<VersionedItem, FrameAndDescriptor>> batch_read_keys(
+    std::vector<ReadVersionOutput> batch_read_keys(
         const std::vector<AtomKey> &keys,
         const std::vector<ReadQuery> &read_queries,
         const ReadOptions& read_options);
 
-    std::vector<std::pair<VersionedItem, FrameAndDescriptor>> batch_read_internal(
+    std::vector<std::variant<ReadVersionOutput, DataError>> batch_read_internal(
```

Comment:
Did you consider std::expected?

Reply:
That's c++ 23, no?

Reply:
Yeh C++23, nice to know the patterns we're working around not existing are coming though.